### PR TITLE
Api phase9 with fixlogs

### PR DESCRIPTION
**Pull request by lrettig**
## Motivation
Same as #2071 but includes a fix for a logging issue (#2092) as well. Want to try running the system tests on this to see if it works around the issue I'm running into in #2071.

### CODE REVIEW

**Comment by lrettig** [comment]
bors try

**Comment by lrettig** [comment]
bors try

**Comment by lrettig** [comment]
bors try

File: log/zap.go [hunk 1]
```diff
@@ -129,57 +127,27 @@ func (l Log) With() FieldLogger {
...
-func (c *coreWithLevel) Enabled(level zapcore.Level) bool {
-	return c.lvl.Enabled(level)
-}
-
-func (c *coreWithLevel) Check(e zapcore.Entry, ce *zapcore.CheckedEntry) *zapcore.CheckedEntry {
-	if !c.lvl.Enabled(e.Level) {
-		return ce
-	}
-	return ce.AddCore(e, c.Core)
+	lgr := l.logger.Named(fmt.Sprintf("%-13s", prefix))
```
Comment (antonlerner) [inline log/zap.go:143]: this completely eliminates dynamic level debugging

File: cmd/node/node.go [hunk 2]
```diff
@@ -445,10 +446,8 @@ func (app *SpacemeshApp) initServices(nodeID types.NodeID,
 
 	app.nodeID = nodeID
 
-	name := nodeID.ShortString()
-
-	lg := log.NewDefault(name).WithFields(nodeID)
-
+	// This base logger must be debug level so that other, derived loggers are not a lower level.
+	lg := log.NewWithLevel(nodeID.ShortString(), zap.NewAtomicLevelAt(zapcore.DebugLevel)).WithFields(nodeID)
```
Comment (antonlerner) [inline cmd/node/node.go:465]: I think this is the only change needed - to have the default logger run with debug level.

**Comment by lrettig** [comment]
bors try

File: log/zap.go (hunk 1 above)
Comment (lrettig) [reply log/zap.go:143]: Yes, that's the idea - see #2090 and #2092 for background. I think we're abusing the log levels here, using them in ways they're not intended to be used, which causes downstream problems. I'll check if I can split this into two, and include just a one- or two-line fix for now, and leave the rest as a separate PR.

**Comment by lrettig** [comment]
bors ping

**Comment by lrettig** [comment]
bors try

File: cmd/node/node.go (hunk 2 above)
Comment (lrettig) [reply cmd/node/node.go:465]: I tested this (https://github.com/spacemeshos/go-spacemesh/pull/2071/commits/d1c68d28339c8da7070d5ed066fe71f917afa7c3), it didn't work. The issue is deeper than this. I explained why in #2090. I don't think we can make a change this superficial here.

**Comment by lrettig** [comment]
> Same as #2071 but includes a fix for a logging issue (#2092) as well. Want to try running the system tests on this to see if it works around the issue I'm running into in #2071.

Confirmed that the failure in #2071 is due to the logging issue, and that the fix in this branch fixes it.

**Comment by lrettig** [comment]
bors try

**Comment by lrettig** [comment]
bors try

**Comment by lrettig** [comment]
bors try-

**Comment by lrettig** [comment]
bors try

**Comment by lrettig** [comment]
bors try

**Comment by lrettig** [comment]
This PR was not meant to be merged, it was just a test to see if adding the logging fixes from #2092 would cause the tests to pass. They're passing. Closing this now, I'll follow up in https://github.com/spacemeshos/go-spacemesh/pull/2071#issuecomment-681021706.

**Pull request closed without merging**